### PR TITLE
fsevents: regression in watching /

### PR DESCRIPTION
**Pull request by vtjnash**
This case got lost by accident while I tried to address review comment https://github.com/libuv/libuv/pull/2082#discussion_r238172936. This prevented the realpath `/` from ever matching a file.

Fixes: https://github.com/nodejs/node/issues/28917
CI: https://ci.nodejs.org/job/libuv-test-commit/1569/

### CODE REVIEW

File: test/test-fs-event.c [hunk 1]
```diff
@@ -393,6 +394,19 @@ static void timer_cb_watch_twice(uv_timer_t* handle) {
   uv_close((uv_handle_t*) handle, NULL);
 }
 
+static void fs_event_cb_close(uv_fs_event_t* handle, const char* filename,
+    int events, int status) {
```
Comment (bnoordhuis) [inline test/test-fs-event.c]: Style nit: can you line up the arguments, one per line?

File: test/test-fs-event.c [hunk 2]
```diff
@@ -458,10 +474,23 @@ TEST_IMPL(fs_event_watch_dir_recursive) {
   r = uv_timer_start(&timer, fs_event_create_files_in_subdir, 100, 0);
   ASSERT(r == 0);
 
+#ifndef _WIN32
+  /* Also try to watch the root directory.
+   * This will be noisier, so we're just checking for any couple events to happen. */
+  r = uv_fs_event_init(loop, &fs_event_root);
+  ASSERT(r == 0);
+  r = uv_fs_event_start(&fs_event_root, fs_event_cb_close, "/", UV_FS_EVENT_RECURSIVE);
```
Comment (bnoordhuis) [inline test/test-fs-event.c]: Long line.

File: test/test-fs-event.c [hunk 3]
```diff
@@ -458,10 +474,23 @@ TEST_IMPL(fs_event_watch_dir_recursive) {
...
+  ASSERT(r == 0);
+#else
+  fs_event_cb_called += 3;
+  close_cb_called += 1;
+#endif
```
Comment (bnoordhuis) [inline test/test-fs-event.c]: Can you add a `(void) &fs_event_root;` to shut up unused variable warnings?

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1583/

**Comment by vtjnash** [comment]
Still a lot of CI problems, but none look related

**Comment by saghul** [comment]
This one is ready to land, right?

**Comment by vtjnash** [comment]
Yes.

**Comment by saghul** [comment]
landed in ae12376d 🚀

**Pull request closed without merging**